File: app/src/main/java/com/xdk78/nimebox/ui/MainActivity.java
package com.xdk78.nimebox.ui;

import android.os.Bundle;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import com.mikepenz.materialdrawer.AccountHeader;
import com.mikepenz.materialdrawer.AccountHeaderBuilder;
import com.mikepenz.materialdrawer.Drawer;
import com.mikepenz.materialdrawer.DrawerBuilder;
import com.mikepenz.materialdrawer.model.PrimaryDrawerItem;
import com.mikepenz.materialdrawer.model.SectionDrawerItem;
import com.xdk78.nimebox.R;

import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity {

    private AccountHeader headerResult;
    private Drawer result;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        MainFragment mainFragment = new MainFragment();
        FragmentManager manager = getSupportFragmentManager();
        manager.beginTransaction().replace(
                R.id.content_main,
                mainFragment
        ).commit();

        new DrawerBuilder().withActivity(this).build();

        headerResult = new AccountHeaderBuilder()
                .withActivity(this)
                .withHeaderBackground(R.drawable.side_nav_bar)
                .build();

        result = new DrawerBuilder()
                .withActivity(this)
                .withToolbar(toolbar)
                .withHasStableIds(true)
                .withActionBarDrawerToggle(true)
                .withActionBarDrawerToggleAnimated(true)
                .withAccountHeader(headerResult)
                .addDrawerItems(
                        new SectionDrawerItem()
                                .withIdentifier(0).withName(R.string.drawer_item_senpai),
                        new PrimaryDrawerItem()
                                .withIdentifier(1).withName(R.string.drawer_item_main),
                        new PrimaryDrawerItem()
                                .withIdentifier(2).withName(R.string.drawer_item_anime),
                        new PrimaryDrawerItem()
                                .withIdentifier(3).withName(R.string.drawer_item_manga),
                        new PrimaryDrawerItem()
                                .withIdentifier(99).withName(R.string.action_settings)

                )
                .withOnDrawerItemClickListener((view, position, drawerItem) -> {

                    if (drawerItem != null) {

                        if (drawerItem.getIdentifier() == 1) {
                            MainFragment mainFragment1 = new MainFragment();
                            FragmentManager manager1 = getSupportFragmentManager();
                            manager1.beginTransaction().replace(
                                    R.id.content_main,
                                    mainFragment1
                            ).commit();
                        } else if (drawerItem.getIdentifier() == 2) {
                            AnimeListFragment animeListFragment = new AnimeListFragment();
                            FragmentManager manager1 = getSupportFragmentManager();
                            manager1.beginTransaction().replace(
                                    R.id.content_main,
                                    animeListFragment
                            ).commit();

                        } else if (drawerItem.getIdentifier() == 3) {
                            MangaListFragment mangaListFragment = new MangaListFragment();
                            FragmentManager manager1 = getSupportFragmentManager();
                            manager1.beginTransaction().replace(
                                    R.id.content_main,
                                    mangaListFragment
                            ).commit();
                        } else if (drawerItem.getIdentifier() == 99) {
                            return false;
                        }

                    }

                    return false;
                })
                .withSavedInstance(savedInstanceState)
                .withShowDrawerOnFirstLaunch(true)
                .build();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        outState = result.saveInstanceState(outState);
        outState = headerResult.saveInstanceState(outState);
        super.onSaveInstanceState(outState);
    }

    @Override
    public void onBackPressed() {
        if (result != null && result.isDrawerOpen()) {
            result.closeDrawer();
        } else {
            super.onBackPressed();
        }
    }
}

File: app/src/main/java/com/xdk78/nimebox/ui/MainActivity.kt
package com.xdk78.nimebox.ui

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import co.zsmb.materialdrawerkt.builders.accountHeader
import co.zsmb.materialdrawerkt.builders.drawer
import co.zsmb.materialdrawerkt.draweritems.badgeable.primaryItem
import com.mikepenz.materialdrawer.AccountHeader
import com.mikepenz.materialdrawer.Drawer
import com.xdk78.nimebox.R
import com.xdk78.nimebox.base.BaseActivity
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : BaseActivity() {

    private lateinit var result: Drawer
    private lateinit var headerResult: AccountHeader

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)
        supportFragmentManager.beginTransaction().replace(R.id.fragment_container, NewsFragment.newInstance()).commit()

        result = drawer {
            hasStableIds = true
            toolbar = this@MainActivity.toolbar
            savedInstance = savedInstanceState
            headerResult = accountHeader {
                background = R.drawable.side_nav_bar
                savedInstance = savedInstanceState
                translucentStatusBar = true
            }

            primaryItem("News") {
                identifier = 0
                onClick { _ ->
                    val fragment = NewsFragment.newInstance()
                    supportFragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).commit()
                    true
                }
            }

            primaryItem("Anime") {
                identifier = 1
                onClick { _ ->
                    val fragment = AnimeListFragment.newInstance()
                    supportFragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).commit()
                    true
                }
            }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun onSaveInstanceState(outState: Bundle?) {
        result.saveInstanceState(outState)
        headerResult.saveInstanceState(outState)
        super.onSaveInstanceState(outState)
    }

    override fun onBackPressed() {
        when {
            result.isDrawerOpen -> result.closeDrawer()
            else -> super.onBackPressed()
        }
    }

}